### PR TITLE
Add test for multiple providers ethereum addresses in ForcePayment

### PR DESCRIPTION
**Pull request by pawelkisielewicz**
It is part of https://github.com/golemfactory/concent/issues/999. Problem was resolved in golem and in this pull request I added test in concent to cover the same case that was tested in golem.

### CODE REVIEW

File: concent_api/core/tests/test_integration_force_payment.py [hunk 1]
```diff
@@ -181,6 +183,69 @@ def test_provider_send_force_payment_with_subtask_results_accepted_where_ethereu
...
+        report_computed_task1 = self._get_deserialized_report_computed_task(
+            timestamp="2018-02-05 10:00:05",
+            task_to_compute=self._get_deserialized_task_to_compute(
+                timestamp="2018-02-05 10:00:00",
+                deadline="2018-02-05 10:00:10",
+                subtask_id=self._get_uuid('1'),
+                price=15000,
+            ))
+
+        provider1_keys = cryptography.ECCx(None)
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_force_payment.py]: `provider1_keys` => `provider_1_keys` same for other variables.

File: concent_api/core/tests/test_integration_force_payment.py [hunk 2]
```diff
@@ -181,6 +183,69 @@ def test_provider_send_force_payment_with_subtask_results_accepted_where_ethereu
...
+        provider_ethereum_public_key = encode_hex(provider1_keys.raw_pubkey)
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_force_payment.py]: Why not using `self._get_provider_ethereum_hex_public_key_different()` ?

File: concent_api/core/tests/test_integration_force_payment.py [hunk 3]
```diff
@@ -181,6 +183,69 @@ def test_provider_send_force_payment_with_subtask_results_accepted_where_ethereu
...
+        Provider  -> Concent:    ForcePayment
+        Concent   -> Provider:   ServiceRefused
+        """
+        report_computed_task1 = self._get_deserialized_report_computed_task(
+            timestamp="2018-02-05 10:00:05",
+            task_to_compute=self._get_deserialized_task_to_compute(
+                timestamp="2018-02-05 10:00:00",
+                deadline="2018-02-05 10:00:10",
+                subtask_id=self._get_uuid('1'),
+                price=15000,
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_force_payment.py]: Please limit the call only to required attributes for this test. For example price is by default `1` and doesn't need to be set to anything extra for this test. It makes it more complicated. Same for other messages.

File: concent_api/core/tests/test_integration_force_payment.py [hunk 4]
```diff
@@ -181,6 +183,69 @@ def test_provider_send_force_payment_with_subtask_results_accepted_where_ethereu
...
+            ),
+        ]
+
+        serialized_force_payment = self._get_serialized_force_payment(
+            timestamp="2018-02-05 12:00:20",
+            subtask_results_accepted_list=subtask_results_accepted_list
+        )
+
+        with freeze_time("2018-02-05 12:00:20"):
+            with mock.patch(
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_force_payment.py]: What is this mock for? The code doesn't reach this place with this scenario.

File: concent_api/core/tests/test_integration_force_payment.py [hunk 5]
```diff
@@ -181,6 +183,69 @@ def test_provider_send_force_payment_with_subtask_results_accepted_where_ethereu
...
+                response = self.send_request(
+                    url='core:send',
+                    data=serialized_force_payment,
+                )
+
+        self._test_response(
+            response,
+            status=200,
+            key=self.PROVIDER_PRIVATE_KEY,
+            message_type=message.concents.ServiceRefused,
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_force_payment.py]: `reason` should be checked too if available.

File: concent_api/core/tests/test_integration_force_payment.py [hunk 6]
```diff
@@ -181,6 +183,69 @@ def test_provider_send_force_payment_with_subtask_results_accepted_where_ethereu
         )
         self._assert_stored_message_counter_not_increased()
 
+    def test_that_if_provider_sends_force_payment_with_two_different_providers_ethereum_accounts_concent_should_refuse(self):
```
Comment (dybi) [inline concent_api/core/tests/test_integration_force_payment.py]: **providers** -> **provider**

File: concent_api/core/tests/test_integration_force_payment.py (hunk 1 above)
Comment (pawelkisielewicz) [reply concent_api/core/tests/test_integration_force_payment.py]: done

File: concent_api/core/tests/test_integration_force_payment.py (hunk 2 above)
Comment (pawelkisielewicz) [reply concent_api/core/tests/test_integration_force_payment.py]: I can do it as well, done :)

File: concent_api/core/tests/test_integration_force_payment.py (hunk 3 above)
Comment (pawelkisielewicz) [reply concent_api/core/tests/test_integration_force_payment.py]: :+1: done

File: concent_api/core/tests/test_integration_force_payment.py (hunk 4 above)
Comment (pawelkisielewicz) [reply concent_api/core/tests/test_integration_force_payment.py]: :+1: It was actually copied from `test_provider_send_correct_force_payment_concent_should_accept`.

File: concent_api/core/tests/test_integration_force_payment.py (hunk 5 above)
Comment (pawelkisielewicz) [reply concent_api/core/tests/test_integration_force_payment.py]: done

File: concent_api/core/tests/test_integration_force_payment.py (hunk 6 above)
Comment (pawelkisielewicz) [reply concent_api/core/tests/test_integration_force_payment.py]: done

File: concent_api/core/tests/test_integration_force_payment.py [hunk 7]
```diff
@@ -181,6 +183,55 @@ def test_provider_send_force_payment_with_subtask_results_accepted_where_ethereu
...
+    def test_that_if_provider_sends_force_payment_with_two_different_provider_ethereum_accounts_concent_should_refuse(self):
+        """
+        Expected message exchange:
+        Provider  -> Concent:    ForcePayment
+        Concent   -> Provider:   ServiceRefused
+        """
+        report_computed_task_1 = self._get_deserialized_report_computed_task(
+            task_to_compute=self._get_deserialized_task_to_compute(
+                subtask_id=self._get_uuid('1'),
+            ))
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_force_payment.py]: Please fix indentation.

File: concent_api/core/tests/test_integration_force_payment.py [hunk 8]
```diff
@@ -181,6 +183,55 @@ def test_provider_send_force_payment_with_subtask_results_accepted_where_ethereu
...
+
+        provider_ethereum_public_key = self._get_provider_ethereum_hex_public_key_different()
+
+        report_computed_task_2 = self._get_deserialized_report_computed_task(
+            task_to_compute=self._get_deserialized_task_to_compute(
+                subtask_id=self._get_uuid('2'),
+                provider_ethereum_public_key=provider_ethereum_public_key,
+            ))
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_force_payment.py]: Same here.

**Pull request merged**